### DISCUSSION TITLE
What exactly is Hare and why is it needed?

### DISCUSSION THREAD

**Discussion by hessio** [Seagate/cortx#597] [Q&A] [answered]
Hare is used to as specified in this [doc](https://github.com/Seagate/cortx-hare#what-hare-does):
1. Configures Motr object store.
2. Starts/stops Motr services.
3. Notifies Motr of service and device faults.

**Comment by mukul-seagate11** [comment, marked as answer]
@hessio, OK but I would say, we can even manage multiple services with systemctl, https://opensource.com/article/20/12/multiple-service-instances-systemctl utility as well

**Reply by andriytk** [reply]
@mukul-seagate11 you are missing the word "distributed", I'm afraid. I'd recommend learning more about Consul and consensus protocols (like Paxos and Raft) - why they are needed and what problems they are solving.

**Comment by hessio** [comment]
Because CORTX is a distributed application (running on many nodes).

It's not easy to start/stop/monitor such applications so we need some application which can keep track of the nodes.

Hare will also generate the cluster configuration from CDF (Cluster Description File).

Hare is built on top of Consul - watch this [video](https://youtu.be/mxeMdl0KvBI) to see what Consul is and that should make more sense why we need Hare.

Hare does not do any recovery, it will notice the problem and inform all instances of the issue. [CORTX-HA](https://github.com/Seagate/cortx-ha#cortx-ha) will take care of failover/failback control flow for affected services and stabilizes them across CORTX cluster.
